process/smartContract/process.go

```diff
@@ -314,6 +324,11 @@ func (sc *scProcessor) doExecuteSmartContractTransaction(
 	snapshot := sc.accounts.JournalLen()
 
 	var vmOutput *vmcommon.VMOutput
+
+	defer func() {
+		sc.vmOutputCacher.Put(txHash, vmOutput, 0)
```

Comment:
this should not happen on a defer func. rather at the end of finishSCExecution when everything is fine. all checks are passed.

Reply:
done.